stdlib/moreLabels.mli

```diff
@@ -1321,12 +1321,97 @@ module Set : sig
       val of_seq : elt Seq.t -> t
       (** Build a set from the given bindings
           @since 4.07 *)
+
+      (** Enumerations of the elements of a set.
+
+          An enumeration is an immutable data structure: its implementation
+          involves no side effects. It represents the sequence of the elements
+          of a set, in increasing order. In addition to enumerating the elements
+          of the sequence (one by one), it supports skipping ahead in the
+          sequence to the point where a certain threshold is reached or
+          exceeded.
+          @since 5.3 *)
+      module Enum : sig
+
+        type set = t
+        (** The type [set] is a synonym for the type of sets.
+            @since 5.3 *)
+
+        type enum
+        (** The type of enumerations. An enumeration represents an increasing
+            sequence of elements of type [elt]. The sequence is increasing
+            according to [Ord.compare]: thus, if [x] appears earlier than [y] in
+            the sequence, then [Ord.compare x y < 0] holds.
+            @since 5.3 *)
+
+        type t = enum
+        (** A synonym for the type [enum].
+            @since 5.3 *)
+
+        val empty : enum
+        (** [empty] is the empty enumeration. It contains zero elements.
+            @since 5.3 *)
+
+        val is_empty : enum -> bool
+        (** [is_empty e] tests whether the enumeration [e] is empty.
+            @since 5.3 *)
+
+        val enum : set -> enum
+        (** [enum s] returns an enumeration of the set [s]. This enumeration
+            contains all of the elements of the set [s], in increasing order.
+            @since 5.3 *)
+
+        val enum_from : elt -> set -> enum
+        (** [enum_from x s] returns an enumeration of the subset of [s]
+            formed of just the elements that are no less than [x].
+            It is equivalent to [from x (enum s)].
+            @since 5.3 *)
```

Comment:
I think that `Myset.Enum.enum` and `Myset.Enum.enum_from` are terrible to read. 

Also `Myset.Enum.elements` (below) is not very good you can't read from the code what you really get. For exemple `Myset.elements` is a list, so if you can remember that you would expect the same from `Myset.Enum.elements`. Of course the best is not to use such confusing names and simply stick to `to_type` functions.

So I would suggest to replace : 

```ocaml
Myset.Enum.enum : set -> t
Myset.Enum.enum_from : elt -> set -> t
Myset.Enum.elements : elt -> set
```
By either: 
```ocaml
Myset.Enum.of_set : set -> t
Myset.Enum.of_set_from : elt -> set -> t
Myset.Enum.to_set : elt -> set
```
Or even:
```ocaml
Myset.Enum.of_set : ?from:elt -> set -> t
Myset.Enum.to_set : elt -> set
```